configuration/pih/htmlforms/section-obgyn-initial.xml

```diff
@@ -544,13 +544,23 @@
 			});
 			
 			beforeSubmit.push(function() {
-			var numErrors = 0;
-			jq('.medicalHistoryText').find('input[type=text]').each(function() {
-				if (!validateFamHistoryText(jq(this)) ) {
-				++numErrors;
-			}
+				var numErrors = 0;
+				jq('.medicalHistoryText').find('input[type=text]').each(function() {
+					if (!validateFamHistoryText(jq(this)) ) {
+					++numErrors;
+					}
+				});
+				return ( numErrors == 0 );
+			});
+
+			jq('#intake-or-followup').change(function() {
+				setRequiredIfVisible('knowing_period_question');
+			});
+			jq('#knowing_period_question').change(function() {
+				setRequiredIfVisible('knowing_period_question');
 			});
-			return ( numErrors == 0 );
+			beforeSubmit.push(function() {
+				return setRequiredIfVisible('knowing_period_question');
```

Comment:
This is more like "validateAsRequiredIfVisible", correct? (admitedly a mouthful of name)

Reply:
Sure, I can rename this.